pos/app/components/molecules/InputComment.tsx

```diff
@@ -0,0 +1,41 @@
+import { useSubmit } from "@remix-run/react";
+import type { WithId } from "common/lib/typeguard";
+import type { OrderEntity } from "common/models/order";
+import { useCallback, useState } from "react";
+import { Input } from "../ui/input";
+
+export const InputComment = ({ order }: { order: WithId<OrderEntity> }) => {
+  const submit = useSubmit();
+  const [descComment, setDescComment] = useState("");
+  const submitComment = useCallback(
+    (servedOrder: OrderEntity, descComment: string) => {
+      const order = servedOrder.clone();
+      order.addComment("master", descComment);
+      submit(
+        { servedOrder: JSON.stringify(order.toOrder()) },
+        { method: "PUT" },
+      );
+    },
+    [submit],
+  );
```

Comment:
コードのお作法の問題なんですが、コンポーネントの中に送信ロジックを書きたくないですね。
routes/_header.master.tsx に submitComment を定義しておいて
InputCommentが props としてその関数を引数に取ってあげるように書き換えたいです。
```suggestion
type props = {
  order: WithId<OrderEntity>;
  mutateOrder: (order: WithId<OrderEntity>) => void; // これをコンポーネントの中で呼び出す
};

export const InputComment = ({ order, mutateOrder }: props) => {
  const [descComment, setDescComment] = useState("");
  // 続きお願いします
```